docs/installation/server-config/security.rst

```diff
@@ -0,0 +1,163 @@
+
+The required ACLs are:
+
+- *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+- *CREATE* permission on the *CLUSTER*.
+- DESCRIBE*, *READ*, *WRITE* and *DELETE* permissions on the *<any>* *TOPIC*.
+- *DESCRIBE* and *READ* permissions  on the *<any>* *GROUP*.
+
+It is still possible to restrict the KSQL user from accessing specific resources using *DENY* ACLs. For example, you can add a
+*DENY* ACL to stop KSQL queries from accessing a topic that contains sensitive data.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Non-Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Non-interactive KSQL clusters will run with much more restrictive ACLs, though it currently requires a little effort to
+work out what ACLs are required. This will be improved in future KSQL releases.
+
+Standard ACLs
+    The KSQL user always requires:
+
+    - *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+    - *DESCRIBE* permission on the *__consumer_offsets* topic.
+
+    If you want KSQL to create internal and sink topics then the KSQL user should also be granted:
+
+    - *CREATE* permission on the *CLUSTER*.
+
+Source topics
+    KSQL users require *DESCRIBE* and *READ* permissions for each source and input topic. The topic should already exist
+    when KSQL is started.
+
+Sink topics
+    KSQL users require *DESCRIBE* and *WRITE* permissions ror each sink and output topic. If the topic does not already
+    exist, the user also requires *CREATE* permissions on the *CLUSTER*.
+
+Change-log and repartition topics
+    The set of changelog and repartitioning topics that KSQL requires depends on the queries being executed. The easiest
+    way to determine the list of required topics is to first run the queries on an open Kafka cluster and list the topics
+    that are created.
+
+    All changelog and repartition topics are prefixed with  ``<value of ksql.service.id property>_query_<query id>_`` where
+    the default of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of *DESCRIBE*, *READ* and *WRITE* permissions for each changelog and repartition *TOPIC*.
+
+    If the KSQL user does not have *CREATE* permissions on the *CLUSTER*, then all changelog and repartition topics must
+    already exist, with the same number of partitions as the source topic, and ``replication.factor`` replicas.
+
+Consumer groups
+    The set of consumer groups that KSQL requires depends on the queries that are being executed. The easiest way to
+    determine the list of consumer groups is to first run the queries on an open Kafka cluster and list the groups created.
+
+    Consumer group names are formatted like ``<value of ksql.service.id property>_query_<query id>``, where the default
+    of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of *DESCRIBE* and *READ* permissions for *GROUP*.
+
+-----------------------------
+Configuring Security for KSQL
+-----------------------------
+
+Here are the general steps to configure KSQL security:
```

Comment:
Sth like this should go into the beginning of this document.  Like, where do you define the security settings? Answer: Security settings (today) only need to go into the KSQL server configuration, and here's where you can find the server configuration file, and changes to that file require a server restart.

@big-andy-coates : Right, today there are no security settings that you'd put into the CLI's (upcoming) config file?

Reply:
i have rearranged and made this the first part of the doc.

Reply:
None that I'm aware of, except for the username/password that @apurvam mentioned for CCloud.

> Answer: Security settings (today) only need to go into the KSQL server configuration

At the moment, Schema Registry requires the use of environment vars, not server config. (PR coming against the master branch, as discussed in team meeting)

Reply:
@big-andy-coates do we need to mention the CCloud username/password in the CCloud docs? (e.g. https://github.com/confluentinc/ksql/blob/4.1.x/docs/installation/server-config/ccloud-ksql.rst)

Reply:
@big-andy-coates @joel-hamill 

There is no need to mention anything additional about CCloud. The username/password for the CLI I referred to were to run against _KSQL Server_ hosted in confluent cloud. This is not something we support right now, so there are no special CLI settings to document for this.